### DISCUSSION TITLE
Custom Variation/Option Creation via API

### DISCUSSION THREAD

**Discussion by Dominic-Moser** [etsy/open-api#1253] [Q&A] [answered]
Currently, creating custom variations through the Etsy API is limited to pre-defined options. This makes it impossible to create variations with custom values that don't fit into these options, like those created using the "Create Your Own" button on the front end.

Request:

Add the ability to create custom variations with user-defined values through the API, mirroring the functionality of the "Create Your Own" feature which is located in the shop interface.

Example Use Case:

A seller offering custom-made products with a wide range of options (e.g., materials, colors, sizes, etc.) that don't align with Etsy's pre-defined values would benefit from this feature.

**Comment by turboturtleD** [comment]
Pretty sure you can already do this.

**Reply by Dominic-Moser** [reply]
Hi @turboturtleD,

Thanks for your reply! I'd really appreciate it if you could clarify how this can be done. I've thoroughly searched the Etsy API documentation, and I haven't been able to find a way to create variations with fully custom values through the API.

If you have a code example, a link to the relevant documentation, or any other information that could help me achieve this, it would be incredibly helpful.

Thanks again for your time and assistance!

**Reply by turboturtleD** [reply, marked as answer]
Without digging back extensively into old code (and discussions that it looks like Etsy may have hidden from their github), I can tell you with relative certainty the endpoint you want is "[updateListingInventory](https://developer.etsy.com/documentation/reference/#operation/updateListingInventory)".

In regards to the property_id for custom values, (products > property_values > property_id) there are two values available for custom properties.  I believe those are 513 and 514.

<img width="290" alt="Screenshot 2024-07-07 at 12 32 16 AM" src="https://github.com/etsy/open-api/assets/78871065/e604b36f-1db1-4122-896c-2de2407eed8e">

This is a challenging endpoint.  If you simply use the getListingInventory endpoint and modify the values that make sense, in my experience and recollection, the request will not succeed.  You need to remove several of the values before re-submitting the data to the update endpoint, it's messy.  If you are already on top of this and working with this endpoint, this should be a piece of cake.  

There are several very old discussions regarding which values need to be stripped for the update to work.  If you need this, I will dig further and try to turn that up.

**Reply by Dominic-Moser** [reply]
@turboturtleD,  thank you so much for taking the time to help me with this issue. I really appreciate the full explanation and links to other discussions you included in your reply. Your solution was exactly what I needed to get this working. Thanks again for your help!

**Reply by pangourelias** [reply]
Thanks, this post was helpful. I also found [this post](https://github.com/etsy/open-api/discussions/1097) helpful.

**Comment by turboturtleD** [comment]
Look here: https://github.com/etsy/open-api/discussions/699
